### ISSUE TITLE
[BUG] AptaTrans: shapes mismatch during pretraning and attention computed on padded zeros

### ISSUE THREAD

**Issue by NennoMP** [gc-os-ai/pyaptamer#177]
There are a few problems in AptaTrans model/training: (i) shape mismatch between predictions and ground-truth labels when pretraining the encoders, and (ii) attention is currently being computed also on padded-zeros. Specifically, the former prevents correct training of the encoders, while the latter is useless from a computational perspective and may also degrade performance.

This is a blocker for #146.
